rust/otap-dataflow/crates/otap/src/object_store.rs

```diff
@@ -211,16 +211,17 @@ mod test {
 
     #[test]
     fn test_get_testdelayed_file_storage() {
-        let storage = StorageType::File {
-            base_uri: "testdelayed:///tmp".to_string(),
-        };
+        let temp_dir = std::env::temp_dir();
+        let base_uri = format!("testdelayed://{}", temp_dir.display());
+        let storage = StorageType::File { base_uri };
         assert!(from_storage_type(&storage).is_ok());
     }
```

Comment:
@cijothomas this is failing on windows. Does std::env::temp_dir() return something like `C:\Users\YourUsername\AppData\Local\Temp`?

I think the reason this test could be failing is that we try to parse this as a url in tests if it starts with `testdelayed`
https://github.com/open-telemetry/otel-arrow/blob/b70ddc7bff3d04142f4a248a985f5557a0bfdbc0/rust/otap-dataflow/crates/otap/src/object_store.rs#L110-L133

We could probably clean up this hack eventually, but in the interim I think it might fix this test if we just strip off the `C:` prefix:

```suggestion
    #[test]
    fn test_get_testdelayed_file_storage() {
        let mut temp_dir = std::env::temp_dir();
        if temp_dir.starts_with("C:") {
            temp_dir = base_uri.strip_prefix("C:").expect("has prefix");
        }
        let base_uri = format!("testdelayed://{}", temp_dir.display());
        let storage = StorageType::File { base_uri };
        assert!(from_storage_type(&storage).is_ok());
    }
```